react/features/base/dialog/components/native/ConfirmDialog.js

```diff
@@ -1,107 +1,171 @@
 // @flow
 
 import React from 'react';
-import { Text, TouchableOpacity } from 'react-native';
+import { View } from 'react-native';
+import Dialog from 'react-native-dialog';
+import { Text } from 'react-native-paper';
 
 import { translate } from '../../../i18n';
 import { connect } from '../../../redux';
-import { StyleType } from '../../../styles';
-import { _abstractMapStateToProps } from '../../functions';
 
-import { type Props as BaseProps } from './BaseDialog';
-import BaseSubmitDialog from './BaseSubmitDialog';
-import { brandedDialog } from './styles';
+import BaseDialog, { type Props as BaseProps } from './BaseDialog';
+import styles from './styles';
 
+
+/**
+ * The type of the React {@code Component} props of
+ * {@link ConfirmDialog}.
+ */
 type Props = BaseProps & {
 
     /**
-     * The color-schemed stylesheet of the feature.
+     * The React {@code Component} children which represents the dialog's body.
      */
-    _dialogStyles: StyleType,
+    children: ?React$Node,
 
     /**
-     * Untranslated i18n key of the content to be displayed.
-     *
-     * NOTE: This dialog also adds support to Object type keys that will be
-     * translated using the provided params. See i18n function
-     * {@code translate(string, Object)} for more details.
+     * Dialog description key for translations.
+     */
+    descriptionKey?: string,
+
+    /**
+     * The i18n key of the text label for the cancel button.
+     */
+    cancelLabel: string,
+
+    /**
+     * The i18n key of the text label for the confirm button.
      */
-    contentKey: string | { key: string, params: Object},
+    confirmLabel: string,
 
     /**
-     * The handler for the event when clicking the 'confirmNo' button.
-     * Defaults to onCancel if absent.
+     * Callback to invoke when cancel button is clicked/pressed.
      */
-    onDecline?: Function,
+    onCancel: Function,
 
+    /**
+     * Callback to invoke when submit button is clicked/pressed.
+     */
+    onSubmit: Function,
+
+    /**
+     * Invoked to obtain translated strings.
+     */
     t: Function
-}
+};
 
 /**
- * Implements a confirm dialog component.
+ * React Component for getting confirmation to stop a file recording session in
+ * progress.
+ *
+ * @augments Component
  */
-class ConfirmDialog extends BaseSubmitDialog<Props, *> {
+class ConfirmDialog<P: Props> extends BaseDialog<Props> {
```

Comment:
Why do we extend from BaseDialog ? What does it provide at this point?

Reply:
this._renderHTML

Reply:
When do we use that? On which specific dialog? I think it'd be best to extract that to a utility function we can then call from any dialog rather than creating an inheritance relationship just for that. Let's give that a try please.